app/routes/blog.$slug.tsx

```diff
@@ -7,19 +8,57 @@ import invariant from "tiny-invariant";
 import { PUBLISH_DATE_FORMATTER } from "~/utils";
 import type { Mdx } from "blog/parser";
 
+export const shortCacheMaxAge = `max-age=28800`; // 8 hour cache
+export const longCacheMaxAge = `max-age=604800`; // One week cache
+
 export const loader = async function ({ params, request }: LoaderArgs) {
   const { slug } = params;
   invariant(typeof slug === "string", "Missing slug");
   const post = await fetch(
     `${new URL(request.url).origin}/resource/get-blog-post/${slug}`
   ).then((response) => response.json() as Promise<Mdx>);
   invariant(post !== undefined, "Could not find post");
-  return {
-    post,
-    url: request.url,
-  };
+  invariant(post.frontmatter.date !== undefined, "Post has no publish date");
+  const publishDateMs = new Date(post.frontmatter.date).getTime();
+  const nowMs = Date.now();
+  const difference = nowMs - publishDateMs;
+  const differenceInDays = Math.floor(difference / (1000 * 60 * 60 * 24));
+  if (process.env.NODE_ENV === "development") {
+    return json({
+      post,
+      url: request.url,
+    });
+  }
+  if (differenceInDays < 4) {
+    return json(
+      {
+        post,
+        url: request.url,
+      },
+      {
+        headers: {
+          "Cache-Control": shortCacheMaxAge,
+        },
+      }
+    );
+  }
+  return json(
+    {
+      post,
+      url: request.url,
+    },
+    {
+      headers: {
+        "Cache-Control": longCacheMaxAge,
+      },
+    }
+  );
```

Comment:
you could shorten this up via:
```typescript
    return json(
      {
        post,
        url: request.url,
      },
      {
        headers: {
          "Cache-Control": (differenceInDays < 4) ? shortCacheMaxAge : longCacheMaxAge,
        },
      }
    );
```

you could even refactor to a func to make it even more legible.

```typescript
function buildHeaders(publishDate: Date) {
  if (process.env.NODE_ENV === "development") {
    return undefined;
  }
  const nowMs = Date.now();
  const difference = nowMs - publishDate.getTime();
  const differenceInDays = Math.floor(difference / (1000 * 60 * 60 * 24));
  return { "Cache-Control": (differenceInDays < 4) ? shortCacheMaxAge : longCacheMaxAge };
}

// in the `loader`
return json(
  {post, url: request.url},
  {headers: buildHeaders(post.frontmatter.date)}
);
```